spec.emu

```diff
@@ -128,9 +128,10 @@ contributors: Luca Casonato, Guy Bedford, Nicolò Ribaudo
             1. Return ~unused~.
           1. Let _module_ be _moduleCompletion_.[[Value]].
           1. <ins>If _state_.[[ModuleReflection]] is *true*, then</ins>
-            1. <ins>If _moduleSourceObject_ is *null*, then</ins>
-              1. <ins>Throw a *TypeError* exception.
             1. <ins>Let _moduleSourceObject_ be _module_.[[ModuleSourceObject]].</ins>
+            1. <ins>If _moduleSourceObject_ is *null*, then</ins>
+              1. <ins>Perform ! Call(_promiseCapability_.[[Reject]], *undefined*, &laquo; a newly created *SyntaxError* object &raquo;).</ins>
```

Comment:
Thanks for catching this, I would still prefer we make it a `TypeError` though, because I think this is more similar to something an import assertion than a missing named export.

Reply:
Ah so we have two because it's one for dynamic import, and one for static syntax. Got it.